google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/application/AppEngineAdminService.java

```diff
@@ -49,8 +73,54 @@ public static AppEngineAdminService getInstance() {
    * @throws IOException if there was a transient error connecting to the API
    */
   @Nullable
-  public abstract Application getApplicationForProjectId(@NotNull String projectId,
-      @NotNull Credential credential) throws IOException, GoogleApiException;
+  public Application getApplicationForProjectId(@NotNull String projectId,
```

Comment:
I'm fine with separating those concerns, but not using inheritance. Put it in the implementation or factor it out some other way.